### PR TITLE
Firefox still has `NodeIterator.detach()`

### PR DESCRIPTION
**Pull request by caugner**
#### Summary

Update the Firefox statement for `NodeIterator.detach()`, indicating that this is still supported.

#### Test results and supporting details

Verified with collector test, see [this comment](https://github.com/mdn/browser-compat-data/issues/20829#issuecomment-3811033154).

#### Related issues

Fixes https://github.com/mdn/browser-compat-data/issues/20829.

### CODE REVIEW

**Comment by caugner** [comment]
> If you make those changes, feel free to merge at your discretion or request re-review. Thank you!

I assume you would expect the same changes for [`Range.detach()`](https://github.com/mdn/browser-compat-data/pull/28936)?

**Comment by ddbeck** [comment]
@caugner yes, that's correct.

**Comment by caugner** [comment]
Done in [29745db](https://github.com/mdn/browser-compat-data/pull/28935/commits/29745dbafb52bf83effafcd16b453a21330e36a4).
